### DISCUSSION TITLE
A few questions about container hot migration

### DISCUSSION THREAD

**Discussion by Carol-code2222** [moby/moby#45067] [Q&A]
### Description

I'm trying a live migration of docker containers, referring to https://criu.org/Docker.
After I run the command:"docker checkpoint create looper ck1" to dump the process in the container, some *.img files generated by criu exist in the directory: /var/lib/docker/containers/$containerID/checkpoints/ck1/, no dump log exists. On the other hand,, when I run the command:"docker start --checkpoint ck1 looper" to restore the process, A restore.log file is stored in the directory：/var/lib/containerd/io.containerd.runtime.v1.linux/moby/$containerID/.

I want to know: 
1. How to record the dump process log? 
2. Where is the restore.log file generated? I cannot find the "-v4 -o restore.log" parameter when the criu command is invoked in the source code of docker-cli, moby, and containerd. 
3. If I want to add some other criu parameters, how can I do it during container live migration?

**Comment by gbakeman** [comment]
On a general sense, container migration in any form or state feels like it could use more attention from the moby/docker teams. What started as an afternoon researching into migrating to the new [containerd image storage](https://docs.docker.com/engine/storage/containerd/), led to looking into migrating any portion of docker which doesn't seem to be really supported. I'm thinking this may be a bigger effort of looking into distributed computing that encapsulates a docker engine environment.
